force-app/main/default/classes/StaticTask.cls

```diff
@@ -1,38 +1,94 @@
 public with sharing class StaticTask {
-  /*This method recieves a string payload with contact data that loks like  
+  /*This method recieves a string payload with contact data that looks like  
 
   'FirstName: firstname, LastName: lastname, Email: ex@mple.com;
    FirstName: firstname, LastName: lastname, Email: ex@mple.com;
    ...
   '
 
-  . Then it parses data and return Map collection where key is meail and value is Contact
-  Feel free to refactorthis class. Don't forget about single responsibility approach when you create a new method.
+  . Then it parses data and return Map collection where key is e-mail and value is Contact
+  Feel free to refactor this class. Don't forget about single responsibility approach when you create a new method.
   */
-  public static Map<String, Contact> extractEmailToContactMap(String payload) {
-    Map<String, Contact> emailToContactMap = new Map<String, Contact>();
-    List<String> contacts = payload.split(';');
-    for (String contactAsString : contacts) {
-      Contact parsedContact = new Contact();
-      List<String> contactSeparatedByComaString = contactAsString.split(',');
-      for (String element : contactSeparatedByComaString) {
-        if (element.contains('FirstName')) {
-          parsedContact.FirstName = element.split(':')[1].replaceAll(' ', '');
+
+    public static Map<String, Contact> extractEmailToContactMap(String payload) {
+
+      List<String> payloadAsStringList = StaticTask.splitPayloadByElementStringList(payload);
+      Map<Integer, List<String>> sObjectFieldsMap = StaticTask.splitStringListByObjectFieldMap(payloadAsStringList);
+      List<Contact> contactList = StaticTask.splitObjectFieldMapByRecords(sObjectFieldsMap);
+      Map<String, Contact> emailAndContactMap = StaticTask.createEmailAndContactMap(contactList);
+
+      return emailAndContactMap;
+  }
+
+
+  public static List<String> splitPayloadByElementStringList (String payload) {
+    return payload.split(';');
+  }
+
+  
+  public static Map<Integer, List<String>> splitStringListByObjectFieldMap (List<String> stringList) {
+    
+    Map<Integer, List<String>> sObjectFieldMap = new Map<Integer, List<String>>();
+    Integer i = 0;
+    
+    for (String element : stringList) {
+      sObjectFieldMap.put(i, element.split(','));
+      i++;
+    }
+
+    return sObjectFieldMap;
+  }
+
+
+  public static List<Contact> splitObjectFieldMapByRecords (Map<Integer, List<String>> integerStringListMap) {
```

Comment:
This method is still pretty complex and hard to read. It has 3 levels of nested statements:

for (String contactString : stringOfContacts) {
for (String con : contactAsList) {
if (con.contains...
Can you try to split it more?